packages/fetch/src/fetch.ts

```diff
@@ -0,0 +1,244 @@
+import {
+  Accessor,
+  createMemo,
+  createResource,
+  ResourceFetcherInfo,
+  ResourceOptions,
+  ResourceReturn
+} from "solid-js";
+import { RequestModifier } from "./modifiers";
+import { fetchRequest, Request } from "./request";
+
+export type FetchArgs = [info: RequestInfo] | [info: RequestInfo, init?: RequestInit];
+
+export type DistributeFetcherArgs<FetcherArgs extends any[], ExtraArgs extends any[]> = [
+  FetcherArgs,
+  ExtraArgs
+] extends [any, any]
+  ?
+      | [...FetcherArgs, ...ExtraArgs]
+      | [Accessor<FetcherArgs | undefined>, ...ExtraArgs]
+      | [...{ [n in keyof FetcherArgs]: Accessor<FetcherArgs[n] | undefined> }, ...ExtraArgs]
+  : never;
+
+export type FetchOptions<Result, InitialValue, FetcherArgs> = ResourceOptions<
+  Result | InitialValue
+> & {
+  fetch?: typeof fetch;
+  request?: (requestContext: RequestContext<Result, FetcherArgs>) => void;
+  responseHandler?: (response: Response) => any;
+  disable?: boolean;
+};
+
+export type FetchReturn<T, I> = [
+  {
+    (): T | I;
+    /** if you are using withAbort(), this will contain a boolean to check if the request was aborted */
+    aborted?: boolean;
+    error: any;
```

Comment:
Ah, I see what are you unhappy about with the types :D
It's probably worth mentioning in the docs, and what's the best path to set the correct types if needed.

Reply:
I think the JSDoc comment will suffice.